clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -982,6 +941,10 @@ boolean maybeExpire() {
             return numAttempts > 0 && isExpired();
         }
 
+        /**
+         * Moves the in-flight acknowledgements for a given partition to incomplete acknowledgements to retry
+         * in the next request.
+         */
         public void moveToIncompleteAcks(TopicIdPartition tip) {
             Acknowledgements acks = inFlightAcknowledgements.remove(tip);
             if (acks != null) {
```

Comment:
Why it needs to check the existent acks from `incompleteAcknowledgements`? It seems `incompleteAcknowledgements` will be empty in this case. By contrast, `moveAllToIncompleteAcks` have no such behavior of checking existent acks.

Reply:
Yeah right, we do not need the additional check here. The `incompleteAcknowledgements` for a given `TopicIdPartition` will be initially empty and be filled utmost once for each `ShareAcknowledgeResponse`. I have removed this check and updated the PR. Thanks!!